src/ui/public/chrome/api/injected_vars.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { cloneDeep } from 'lodash';
+import { InjectedMetadataStartContract } from '../../../../core/public/injected_metadata';
+
+let newPlatformInjectedVars: InjectedMetadataStartContract;
+
+export function __newPlatformInit__(instance: InjectedMetadataStartContract) {
+  if (newPlatformInjectedVars) {
+    throw new Error('ui/chrome/api/injected_vars is already initialized');
+  }
+
+  newPlatformInjectedVars = instance;
+}
+
+export function initChromeInjectedVarsApi(chrome: any) {
```

Comment:
Yeah, but defining the type as anything other than `any` or something like

```ts
initChromeInjectedVarsApi(chrome: { getInjected?: (name: string, defaultValue: any) => any })
```
produces errors because you can't just add properties to objects in TS, and since defining the signature for `getInjected` twice felt redundant I went for any here. It wouldn't spread any outside of this module at least, unless we went and converted the chrome in its current form, which I don't think we'll ever do.

Reply:
What about `initChromeInjectedVarsApi(chrome: {[key: string]: any})`? That at least ensures it is given an object and shouldn't complain about setting new properties.